comp/core/workloadmeta/collectors/internal/process/process_collector.go

```diff
@@ -0,0 +1,133 @@
+
+// Package process implements the local process collector for
+// Workloadmeta.
+package process
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"go.uber.org/fx"
+
+	workloadmeta "github.com/DataDog/datadog-agent/comp/core/workloadmeta/def"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/errors"
+	processwlm "github.com/DataDog/datadog-agent/pkg/process/metadata/workloadmeta"
+	"github.com/DataDog/datadog-agent/pkg/status/health"
+	"github.com/DataDog/datadog-agent/pkg/util/flavor"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	collectorID       = "local-process-collector"
+	cacheValidityNoRT = 2 * time.Second
+	componentName     = "workloadmeta-process"
+)
+
+type collector struct {
+	id      string
+	store   workloadmeta.Component
+	catalog workloadmeta.AgentType
+
+	processDiffCh <-chan *processwlm.ProcessCacheDiff
+}
+
+// NewCollector returns a new docker collector provider and an error
+func NewCollector() (workloadmeta.CollectorProvider, error) {
+	return workloadmeta.CollectorProvider{
+		Collector: &collector{
+			id:      collectorID,
+			catalog: workloadmeta.NodeAgent,
+		},
+	}, nil
+}
+
+// GetFxOptions returns the FX framework options for the collector
+func GetFxOptions() fx.Option {
+	return fx.Provide(NewCollector)
+}
+
+func (c *collector) Start(ctx context.Context, store workloadmeta.Component) error {
+	if !config.Datadog().GetBool("language_detection.enabled") && flavor.GetFlavor() != flavor.DefaultAgent {
+		return errors.NewDisabled(componentName, "core agent language detection not enabled")
+	}
+	c.store = store
+	c.processDiffCh = processwlm.GetSharedWorkloadMetaExtractor(config.SystemProbe).ProcessCacheDiff()
+
+	go c.stream(ctx)
+
+	return nil
+}
+
+func CollectorEventsFromProcessDiff(creations []*processwlm.ProcessEntity, deletions []*processwlm.ProcessEntity) []workloadmeta.CollectorEvent {
+	events := make([]workloadmeta.CollectorEvent, 0, len(creations)+len(deletions))
+
+	for _, creation := range creations {
+		events = append(events, workloadmeta.CollectorEvent{
+			Type: workloadmeta.EventTypeSet,
+			Entity: &workloadmeta.Process{
+				EntityID: workloadmeta.EntityID{
+					Kind: workloadmeta.KindProcess,
+					ID:   strconv.Itoa(int(creation.Pid)),
+				},
+				ContainerID:  creation.ContainerId,
+				NsPid:        creation.NsPid,
+				CreationTime: time.UnixMilli(creation.CreationTime),
+				Language:     creation.Language,
+			},
+			Source: workloadmeta.SourceLocalProcessCollector,
+		})
+	}
+
+	for _, deletion := range deletions {
+		events = append(events, workloadmeta.CollectorEvent{
+			Type: workloadmeta.EventTypeUnset,
+			Entity: &workloadmeta.Process{
+				EntityID: workloadmeta.EntityID{
+					Kind: workloadmeta.KindProcess,
+					ID:   strconv.Itoa(int(deletion.Pid)),
+				},
+			},
+			Source: workloadmeta.SourceLocalProcessCollector,
+		})
+	}
+
+	return events
+}
+
+func (c *collector) stream(ctx context.Context) error {
```

Comment:
stream method should not be returning anything. 
it should only use channels and logging.

Reply:
you can check the [stream method of the docker collector](https://github.com/DataDog/datadog-agent/blob/433dfce57428d08591e1ab7d2a497c8752ad9fa7/comp/core/workloadmeta/collectors/internal/docker/docker.go#L140-L178)

Reply:
Done